File: 12. Integer to Roman Medium/solution.java
class solution {
    public static void main(String[] args) {
        System.out.println(intToRoman(3749));
        System.out.println(intToRoman(58));
        System.out.println(intToRoman(1994));
    }

    public static String intToRoman(int num) {
        int[] nums = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] romans = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};

        StringBuilder result = new StringBuilder();

        // Iterate through associations
        for (int i = 0; i < nums.length; i++) {
            int k = nums[i];
            String v = romans[i];
            while (num >= k) {
                result.append(v);
                num -= k;
            }
        }

        return result.toString();
    }
}

File: 12. Integer to Roman Medium/solution.ts
function intToRoman(num: number): string {
    const romanNumerals: [number, string][] = [
        [1000, "M"],
        [900, "CM"],
        [500, "D"],
        [400, "CD"],
        [100, "C"],
        [90, "XC"],
        [50, "L"],
        [40, "XL"],
        [10, "X"],
        [9, "IX"],
        [5, "V"],
        [4, "IV"],
        [1, "I"],
    ];

    let result: string = '';

    for (let [value, roman] of romanNumerals) {
        while (num >= value) {
            result += roman;
            num -= value;
        }
        if (num === 0) break;
    }

    return result;
}   

console.log(intToRoman(3749))
console.log(intToRoman(58))
console.log(intToRoman(1994))